src/main/java/Duke.java

```diff
@@ -1,10 +1,139 @@
+import exception.DukeException;
+import task.Command;
+import task.Task;
+import task.TaskManager;
+import text.ErrorMessage;
+import text.Print;
+
+import java.io.IOException;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+    private static Command instruction;
+    private static String description;
+    private static final TaskManager taskManager = new TaskManager();
+
+
+    public static void main(String[] args) throws IOException {
+        TaskManager.loadSavedFile();
+        Print.printWelcomeMessage();
+
+        while (true) {
+            String userInput = askForInput();
+
+            try {
+                checkTypeOfInstruction(userInput);
+            } catch (DukeException e) {
+                ErrorMessage.checkTypeOutOfBoundsException();
+                continue;
+            }
+
+            try {
+                splitInput(userInput);
+            } catch (DukeException e) {
+                ErrorMessage.splitInputOutOfBoundsException();
+                continue;
+            }
+
+            if (instruction.isCommand()) {
+                executeCommand();
+            } else if (instruction.isTask()) {
+                Task task = taskManager.createTask(instruction, description);
+                taskManager.addToList(task);
+            }
+
+            if (instruction == Command.LIST || instruction == Command.DONE || instruction == Command.DELETE) {
+                continue;
+            }
+
+            Print.printSingleTask(taskManager, TaskManager.getTaskCount()-1);
+            Print.printNoOfTasks(taskManager);
+            TaskManager.saveFile();
+        }
+    }
+
+
+    private static String askForInput() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Please enter a task:");
+
+        Scanner input = new Scanner(System.in);
+        String userInput = input.nextLine();
+        return userInput;
+    }
+
+    public static void checkTypeOfInstruction(String input) throws DukeException {
```

Comment:
Seems like you're setting (instead of just checking) the instruction enum for the corresponding truth tables of the 4 boolean conditions you assigned for each instruction. Also if you call them instructions - I think it'll be less confusing if you use Instruction instead of Command as the name for the enums. Also consider renaming the method to maybe setInstructionEnumType?